code-challenges.test.js

```diff
@@ -10,37 +12,104 @@
 // Run the file with the following command: $ yarn jest
 
+
+
+//   ***Hate to end on such a low note, but my family has decided to pay me a surprise visit and now I barely have time to think. Apologies for the subpar work on this one.***
+
+
 // --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
 // a) Create a test with an expect statement using the variable provided.
 
+// input: an array of objects
+// output: an array with a sentence about each person with their name capitalized.
+// Pseudo: Map across the array, using a regular expression powered .replace method to capitalize the names. after that, need to stick the output in a concatenated string and return those as an array.
+
+
+
+
+
+describe('heartOfGold', () => {
+  it('return an array with a sentence about each person with their name capitalized', () => {
+    expect(heartOfGold(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  });
+});
+
 const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
+
+const heartOfGold = (towel) => {
+  const properPeople = towel.map(person => {
+    const nominalName = person.name.replace(/\b\w/g, char => char.toUpperCase())
+    return { name: nominalName, occupation: person.occupation }
+  })
+
+  const noStringsAttached = properPeople.map(person => `${person.name} is ${person.occupation}.`)
+  return noStringsAttached
+}
+
+console.log(heartOfGold(people))
+
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
 // b) Create the function that makes the test pass.
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
+// input: a mixed data array
+// output: an array of only the REMAINDERS of the numbers when divided by three
+// Pseudo: .filter to extract the numbers from the array, then .map shenanigans to do the math-y bits.
+
 // a) Create a test with an expect statement using the variables provided.
 
+describe('booMath', () => {
+  it('return an array of only the REMAINDERS of the numbers when divided by 3 ', () => {
+    expect(booMath(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(booMath(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  });
+});
+
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
+const booMath = (input) => {
+  const onlyNums = input.filter(bruh => typeof bruh === "number")
+  remainders = onlyNums.map(brainHurts => brainHurts % 3)
```

Comment:
Nice! Just make sure you declare your variables otherwise JavaScript will make them global.